### PR TITLE
RUMM-556 Add Room use case in the sample app

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

The goal of this PR was to provide integration support for Room Persistence Library. In the end we could not find any reliable backdoor in order to get a callback whenever a DB corruption is detected and we will have to rely on the `DatadogUncaughtExceptionHandler` to catch and deliver all the unhandled possible exceptions thrown by Room framework code.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: buildSrc/src/main/kotlin/com/datadog/gradle/Dependencies.kt [hunk 1]
```diff
@@ -168,6 +169,8 @@ object Dependencies {
             "com.github.bumptech.glide:okhttp3-integration:${Versions.Glide}"
         )
 
+        const val Room = "androidx.room:room-runtime:${Versions.Room}"
+        const val RoomKapt = "androidx.room:room-compiler:${Versions.Room}"
```
Comment (xgouchet) [inline buildSrc/src/main/kotlin/com/datadog/gradle/Dependencies.kt:173]: This one should be defined in the `AnnotationProcessors` object below, and can simply be named `Room`

File: sample/kotlin/src/room/kotlin/com/datadog/android/sample/data/db/LogRoom.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import java.util.UUID
+
+@Entity(tableName = DatadogDbContract.Logs.TABLE_NAME)
+data class LogRoom(
+    @PrimaryKey
+    @ColumnInfo(name = "id")
```
Comment (xgouchet) [inline sample/kotlin/src/room/kotlin/com/datadog/android/sample/data/db/LogRoom.kt:17]: Why isn't this one defined as a constant ?

**Review by xgouchet — CHANGES_REQUESTED** [review]
Nice work, just a couple of minor changes

File: sample/kotlin/src/room/kotlin/com/datadog/android/sample/data/db/LogRoom.kt (hunk 2 above)
Comment (mariusc83) [reply sample/kotlin/src/room/kotlin/com/datadog/android/sample/data/db/LogRoom.kt:17]: Oh...I missed that ;)

**Review by xgouchet — APPROVED** [review]

**Pull request merged**